zrml/prediction-markets/src/lib.rs

```diff
@@ -1225,6 +1235,15 @@ mod pallet {
                     },
                 );
 
+                if let Ok(weight) = open {
+                    total_weight = total_weight.saturating_add(weight);
+                } else {
```

Comment:
I don't understand this else part, so when open_market() fails then it goes to else here but then technically on_initialize() is failed? and why that means we need to charge for worst case? Same question for resolution and close cases.

Reply:
Yeah, that's a good question. This is a conservative approach. We don't know when this failed [here](https://github.com/zeitgeistpm/zeitgeist/blob/30b7191666809c5f40293b0e815c9336d3a9ec91/zrml/prediction-markets/src/lib.rs#L2194-L2195) or [here](https://github.com/zeitgeistpm/zeitgeist/blob/30b7191666809c5f40293b0e815c9336d3a9ec91/zrml/prediction-markets/src/lib.rs#L2205-L2206). It could be in the last or first for-loop-iteration. The corrected weight is only returned in the `Ok(weight)` case, but not in early return errors. So it could be, that `market_status_manager` failed at the last for-iteration (thus taking lots of processing time).